### PR TITLE
change mimedb store stringtable to orderedtable

### LINKED ISSUE #18060: mimetypes m.getExt("application/json") returns ndjson
Function `echo` outputs the wrong string.

### Example
```nim
import mimetypes
var m = newMimetypes()
echo m.getExt("application/json")
```

### Current Output

```
ndjson
```

### Expected Output
```
json
```

### Possible Solution

* in `func newMimetypes` changes to `result.mimes = mimes.newOrderedTable()`

### Additional Information  
multiple extensions could registered to same mime type , however the primary type should take high order in `getExt` proc  
https://www.iana.org/assignments/media-types/media-types.xhtml

```
$ nim -v
1.4.6 and devel(1.5.1 15586c7a7a54f9d573eed9ec4ec90994e1e0c483)
```

### PR DESCRIPTION
**Pull request by bung87**
fix #18060

### CODE REVIEW

**Comment by juancarlospaco** [comment]
I do not think this is the right solution, and it is a breaking change, just fix the `"json"` as workaround.

I think a new API is needed, that uses `{ string: seq[string] }`.

**Comment by bung87** [comment]
I'v considered that , but I dont think that's a break change , the end user might use `ndjson` but that's definitely wrong, I can't " just fix the "json" as workaround" , a new api getExts make sense, but remain current getExt implementation is bad.

**Comment by juancarlospaco** [comment]
Pre-existent implementation is bad.
Fix it by using values like `"json,ndjson"` then user can `value.split ','` to get `seq`, and keeps being `{ string: string }`.

**Comment by bung87** [comment]
> Pre-existent implementation is bad.
> Fix it by using values like `"json,ndjson"` then user can `value.split ','` to get `seq`, and keeps being `{ string: string }`.

for this case I think just introduce new api `getExts` that's much better, returns `"json,ndjson"` still breaking end user code.

**Comment by Araq** [comment]
How is this a breaking change? The field is not public, the implementation was not exposed and it cannot be iterated over.

**Comment by bung87** [comment]
will break user code `if getExt(mime) == "ndjson" : do something`

**Comment by planetis-m** [comment]
Previous attempt: #12368 (Can you redo the optimization in `getMimetype` proc?)

**Comment by bung87** [comment]
> getMimetype

Could you explain more ? I dont see any problem in `getMimetype`

**Comment by planetis-m** [comment]
Doesn't really matter but if you see the previous PR diff, I had reworked `getMimetype` a bit, might worth salvaging.

**Comment by timotheecour** [comment]
CI fails for tests/effects/tstrict_funcs_imports.nim: ../../lib/pure/mimetypes.nim(1903, 6) Error: 'newMimetypes' can have side effects

`func newMimetypes*(): MimeDB =` now doesn't compile because of newOrderedTable, and changing newOrderedTable to func would fail because of 
```
proc `[]=`*[A, B](t: OrderedTableRef[A, B], key: A, val: sink B) =
```
which can't be changed to func (even though it should be func) because of `--experimental:strictFuncs`

These should all be func's, the problem is `--experimental:strictFuncs` which is IMO flawed, refs https://github.com/nim-lang/Nim/issues/16305

**Comment by Araq** [comment]
> These should all be func's, the problem is --experimental:strictFuncs which is IMO flawed, refs #16305

But in reality the `TableRef` versions are flawed as they obscure the mutation aspect.

File: lib/pure/mimetypes.nim [hunk 1]
```diff
@@ -1903,7 +1903,7 @@ const mimes* = {
 func newMimetypes*(): MimeDB =
   ## Creates a new Mimetypes database. The database will contain the most
   ## common mimetypes.
-  result.mimes = mimes.newStringTable()
+  result.mimes = mimes.newOrderedTable()
```
Comment (Araq) [inline lib/pure/mimetypes.nim]:
```suggestion
  {.cast(noSideEffect).}:
    result.mimes = mimes.newOrderedTable()
```

**Comment by timotheecour** [comment]
> But in reality the TableRef versions are flawed as they obscure the mutation aspect.

it just shows that conflating {.noSideEffect.} with transitive const-ness [1] is flawed, there should be a different pragam for enforcing transitive constness (eg `proc fn(a {.transitiveconst.}: Foo)` or `proc fn(a: 
 Foo) {.transitiveconst.}`); these are different concepts.

`{.cast(noSideEffect).}:` is just band-aid that masks real side effects, making the code even less sound that it was before `--experimental:strictFuncs`

in D, `const` being transitive was a bad design decision [2], preventing logical const, caching fields (for lazy initialization) etc, and is not used much in library code [3] because of its viral nature; nim, until the introduction of `--experimental:strictFuncs`, had made the right move away from transitive const.

[1] note: in all this, const is taken with the meaning of the one in C, C++, D, not in the sense of "compile time"; note that immutable has yet another meaning (eg in D), so I didn't use that term here

[2] https://digitalmars.com/d/archives/digitalmars/D/Transitive_const_sucks_58111.html#N58111, https://forum.dlang.org/post/mailman.591.1520271555.3374.digitalmars-d-announce@puremagic.com and other articles, eg https://softwareengineering.stackexchange.com/questions/97207/what-does-c-do-better-than-d regarding transitive const

[3] most of the uses of const in D's stdlib are for strings or array buffers, not for classes/structs, because transitive const would cause more trouble that it'd solve; this is especially true in generic code

**Comment by Araq** [comment]
> it just shows that conflating {.noSideEffect.} with transitive const-ness [1] is flawed, there should be a different pragam for enforcing transitive constness (eg proc fn(a {.transitiveconst.}: Foo) or proc fn(a: Foo) {.transitiveconst.}); these are different concepts.

Well, arguably `.transitiveconst` should be the default, leaving us with `proc fn(x: mut ref T)` but `ref T` is over-used in Nim anyway, and `T` vs `var T` is the existing, elegant solution.

> {.cast(noSideEffect).}: is just band-aid that masks real side effects, making the code even less sound that it was before --experimental:strictFuncs

Maybe, but I'm not sure. Because the existing "no globals allowed" noSideEffect had very few benefits, it simply doesn't find real bugs in practice. The new interpretation of "noSideEffect" at least gives Nim's optimizer benefits and that is objective.

**Pull request merged**